PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['res.company']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.users.log']

Total Changes: 12

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_presence/models/hr_employee.py
+++ b/addons/hr_presence/models/hr_employee.py
@@ -11,7 +11,7 @@ _logger = logging.getLogger(__name__)
 
 
 class HrEmployeeBase(models.AbstractModel):
-    _inherit = ['hr.employee.base']
+    _inherit = 'hr.employee.base'
 
     email_sent = fields.Boolean(default=False)
     ip_connected = fields.Boolean(default=False)

--- a/addons/hr_presence/models/hr_employee_base.py
+++ b/addons/hr_presence/models/hr_employee_base.py
@@ -3,7 +3,7 @@ from odoo import api, fields, models
 
 
 class HrEmployeeBase(models.AbstractModel):
-    _inherit = ["hr.employee.base"]
+    _inherit = "hr.employee.base"
 
     @api.depends("user_id.im_status", "hr_presence_state_display")
     def _compute_presence_state(self):

--- a/addons/hr_presence/models/ir_websocket.py
+++ b/addons/hr_presence/models/ir_websocket.py
@@ -9,7 +9,7 @@ from odoo.addons.bus.websocket import wsrequest
 
 
 class IrWebsocket(models.AbstractModel):
-    _inherit = ['ir.websocket']
+    _inherit = 'ir.websocket'
 
     def _update_bus_presence(self, inactivity_period, im_status_ids_by_model):
         super()._update_bus_presence(inactivity_period, im_status_ids_by_model)

--- a/addons/hr_presence/models/res_company.py
+++ b/addons/hr_presence/models/res_company.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResCompany(models.Model):
-    _inherit = ['res.company']
+    _inherit = 'res.company'
 
     hr_presence_last_compute_date = fields.Datetime()

--- a/addons/hr_presence/models/res_config_settings.py
+++ b/addons/hr_presence/models/res_config_settings.py
@@ -3,7 +3,7 @@ from odoo import models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     def create(self, vals):
         configs = super().create(vals)

--- a/addons/hr_presence/models/res_users_log.py
+++ b/addons/hr_presence/models/res_users_log.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models
 
 
 class ResUsersLog(models.Model):
-    _inherit = ['res.users.log']
+    _inherit = 'res.users.log'
 
     create_uid = fields.Integer(index=True)
     ip = fields.Char(string="IP Address")
